federationexample/devicegqlserver/main.go

```diff
@@ -0,0 +1,92 @@
+
+
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+	
+	"google.golang.org/grpc"
+
+	"github.com/samsarahq/thunder/federation"
+	"github.com/samsarahq/thunder/graphql/schemabuilder"
+	"github.com/samsarahq/thunder/thunderpb"
+
+)
+
+func schema() *schemabuilder.Schema {
+	schema := schemabuilder.NewSchemaWithName("device")
+
+
+	type Location struct {
+		Latitude float64
+		Longitude float64
+	}
+	location := schema.Object("Location", Location{}, 
+		schemabuilder.FetchObjectFromKeys(func(ctx context.Context, args struct{ Keys []*Location }) ([]*Location) {
+			return args.Keys
+		}),
+	)
+
+	location.FieldFunc("street", func(ctx context.Context, location *Location) (string , error) {
+		return "My street address", nil
+	})
+
+
+
+	type Device struct {
+		Id int64
+		Name string
+	}
+	type DeviceKey struct {
+		Id int64
+	}
+		device := schema.Object("Device", Device{},
```

Comment:
As a general comment, I'm a bit confused about why this fetches `device` (singular), when FetchObjectFromKeys() return multiple devices.

Reply:
Yup, so if we make a db fetch in `FetchObjectFromKeys` we want it to be a single db queries for all the devices instead of separate db fetches for every device. To do this we need the  FetchObjectFromKeys to take a set of devices so it has enough information to make a single db request.